### ISSUE TITLE
Slowness of Scene() creation

### ISSUE THREAD

**Issue by pnuu** [pytroll/satpy#46]
When creating the Scene object with the base_dir given, the creation is very, very slow, if there are plenty of files in the directory. For example, I have now 6337 files in my data directory, the scene creation using command
`glbl = Scene(platform_name="Meteosat-10", sensor="seviri", reader="hrit_msg", start_time=tslot, base_dir="/path/to/data")`
takes 42.7 seconds. The given parameters should already contain all the needed information to get the correct set of files without opening all the existing files. In comparison
`glbl = Scene(platform_name="Meteosat-10", sensor="seviri", reader="hrit_msg", start_time=tslot, filenames=glob("/path/to/data/*201409051145*"))`
takes only 0.8 seconds.

I haven't yet looked under the hood, but maybe it would be possible to add a shortcut for the cases where the files can be selected based on the given data. Or if it's a smaller set, handle the data which need to be opened to get the exact time (etc) as a special case.

**Comment by davidh-ssec** [comment]
@pnuu @mraspaud I saw that you guys discussed this on slack a little bit, but I'll add some thoughts here.

Agreed this is not good that it takes so long to automatically find files. What if we added logic for saying:

    if start_time is not None and 'start_time' in filename_info:
        # filter by start_time

And the same for `end_time`. I think there would have to be some keyword to turn this off if it is on by default.

Thoughts?

**Comment by davidh-ssec** [comment]
Also how would we handle more complicated cases like the VIIRS SDRs (iirc) that have one date and two times to represent start time and end time?

**Comment by mraspaud** [comment]
what we do in other parts of pytroll is to fix the end time to be always after the start time.

**Comment by davidh-ssec** [comment]
Ok thoughts on the following implementation that I will implement:

1. Add `filter_filenames=True` to the `YAMLFileReader` class (*not* `AbstractYAMLReader`).
    I would like to make it specified in the YAML config, but the keywords coming from the Scene could override it.
2. Modify the filename selecting methods in the reader to get filename information as fast as possible.
3. Filter based on `start_time` and `end_time` in filename info.

@mraspaud I've reviewed the code that would need to be modified and I found a small issue. The ReaderFinder now depends on `create_filehandlers` being defined in the Reader class but it isn't defined in the `AbstractYAMLReader`. Maybe a new abstractmethod?
